aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,660 @@
+    struct Proposal has copy, drop, store {
+        /// Name of the proposal, limiting to 64 chars
+        name: String,
+        /// Description of the proposal, limiting to 512 chars
+        description: String,
+        /// The name of function to be executed
+        function_name: String,
+        /// The function arguments to be executed
+        function_args: PropertyMap,
+        /// The start time of the voting
+        start_time_sec: u64,
+        /// Proposal results, unresolved, passed, not passed
+        resolution: u8,
+        /// final voting count of yes votes
+        final_yes_votes: u64,
+        /// final voting count of no votes
+        final_no_votes: u64,
+    }
+
+    struct ProposalVotingStatistics has key {
+        proposals: Table<u64, VotingStatistics>,
+    }
+
+    struct VotingStatistics has store {
+        /// Total yes votes
+        total_yes: u64,
+        /// Total no notes
+        total_no: u64,
+        /// Token voted yes
+        yes_votes: BucketTable<TokenId, address>, // address is the original voter's address for keeping a record of who voted
+        /// Token voted no
+        no_votes: BucketTable<TokenId, address>,
+    }
+
+    /// Creator creates a DAO on the platform
+    public entry fun create_dao(
+        admin: &signer,
+        name: String,
+        threshold: u64,
+        voting_duration: u64,
+        voting_token_collection_creator: address, // this is the creator address of goverance token
+        collection_name: String,
+        min_required_proposer_voting_power: u64,
+    ) acquires OwnedDAO {
+
+        // create a resource account
+        let seed = bcs::to_bytes(&name);
+        vector::append(&mut seed, bcs::to_bytes(&voting_token_collection_creator));
+        vector::append(&mut seed, bcs::to_bytes(&collection_name));
+
+        let (res_signer, res_cap) = account::create_resource_account(admin, seed);
+        let src_addr = signer::address_of(admin);
+
+        assert!(string::length(&name) < 128, error::invalid_argument(ESTRING_TOO_LONG));
+
+        move_to(
+            &res_signer,
+            DAO {
+                name,
+                resolve_threshold: threshold,
+                governance_token: GovernanceToken { creator: voting_token_collection_creator, collection: collection_name },
+                voting_duration,
+                min_required_proposer_voting_power,
+                next_proposal_id: 0,
+                dao_signer_capability: res_cap,
+                admin: src_addr,
+            },
+        );
+        move_to(
+            &res_signer,
+            Proposals {
+                proposals: table::new()
+            }
+        );
+
+        move_to(
+          &res_signer,
+            ProposalVotingStatistics {
+                proposals: table::new()
+            }
+        );
+
+        let dao_addr = signer::address_of(&res_signer);
+        if (!exists<OwnedDAO>(src_addr)) {
```

Comment:
Can we separate this into a private function to improve readability?

Reply:
done